### ISSUE TITLE
ivy: SPC h r throws error 

### ISSUE THREAD

**Issue by nixmaniack** [syl20bnr/spacemacs#6513]
#### Description :octocat:
Errors out when selecting document to read using `SPC h r`

#### Reproduction guide :beetle:
- Start Emacs
- Hit `SPC h r` and select a candidate to read documentation

*Observed behaviour:* :eyes: :broken_heart:
Throws error:
```
if: Wrong type argument: stringp, ("Spacemacs documentation" . "DOCUMENTATION.org")
```

*Expected behaviour:* :heart: :smile:
Open selected documentation

#### System Info :computer:
- OS: darwin
- Emacs: 25.1.50.1
- Spacemacs: 0.105.21
- Spacemacs branch: develop (rev. a650877)
- Graphic display: t
- Distribution: spacemacs
- Editing style: hybrid
- Completion: ivy
- Layers:
```elisp
(ivy restclient
     (auto-completion :variables auto-completion-enable-snippets-in-popup t auto-completion-enable-sort-by-usage t auto-completion-enable-help-tooltip nil auto-completion-private-snippets-directory "~/.spacemacs.d/snippets/" :disabled-for org erc)
     semantic better-defaults emacs-lisp
     (git :variables git-gutter-use-fringe t git-magit-status-fullscreen nil)
     github markdown yaml shell
     (spell-checking :variables spell-checking-enable-by-default nil spell-checking-enable-auto-dictionary t enable-flyspell-auto-completion t)
     syntax-checking twitter
     (elfeed :variables rmh-elfeed-org-files
             (list "~/.emacs.d/private/elfeed.org"))
     lua
     (python :variables python-enable-yapf-format-on-save t python-test-runner
             '(nose pytest))
     ruby ruby-on-rails
     (clojure :variables clojure-enable-fancify-symbols t)
     go shell-scripts html javascript docker spacemacs-layouts window-purpose
     (mu4e :variables mu4e-installation-path "/usr/local/share/emacs/site-lisp/mu/mu4e/" mu4e-enable-mode-line t mu4e-enable-notifications t)
     nix-mu4e colors deft finance
     (ibuffer :variables ibuffer-group-buffers-by 'projects)
     imenu-list
     (version-control :variables version-control-diff-tool 'diff-hl)
     ansible emoji ranger prodigy osx typography
     (theming :variables theming-headings-inherit-from-default 'all theming-headings-same-size 'all theming-headings-bold 'all)
     org
     (plantuml :variables org-plantuml-jar-path
               (expand-file-name "~/bin/plantuml.jar")
               puml-plantuml-jar-path
               (expand-file-name "~/bin/plantuml.jar"))
     command-log erc pdf-tools bh-org-mode
     (evil-snipe :variables evil-snipe-enable-alternate-f-and-t-behaviors t))
```

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (wrong-type-argument stringp ("Spacemacs documentation" . "DOCUMENTATION.org"))
  string=(("Spacemacs documentation" . "DOCUMENTATION.org") "CONTRIBUTING.org")
  (if (string= candidate "CONTRIBUTING.org") (concat spacemacs-start-directory candidate) (concat spacemacs-docs-directory candidate))
  (let ((file (if (string= candidate "CONTRIBUTING.org") (concat spacemacs-start-directory candidate) (concat spacemacs-docs-directory candidate)))) (cond ((equal (file-name-extension file) "md") (condition-case-unless-debug nil (with-current-buffer (find-file-noselect file) (gh-md-render-buffer) (spacemacs/kill-this-buffer)) (find-file file))) ((equal (file-name-extension file) "org") (spacemacs/view-org-file file "^" (quote all))) (t (find-file file))))
  ivy-spacemacs-help//documentation-action-open-file(("Spacemacs documentation" . "DOCUMENTATION.org"))
  ivy-call()
  ivy-read("Spacemacs Documentation: " (("Spacemacs documentation" . "DOCUMENTATION.org") ("How t
```

**Comment by bixuanzju** [comment]
I am still having this issue

#### System Info :computer:
- OS: darwin
- Emacs: 25.0.95.1
- Spacemacs: 0.105.21
- Spacemacs branch: develop (rev. 9499e24)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
(ivy
 (auto-completion :variables auto-completion-enable-sort-by-usage t)
 emacs-lisp
 (git :variables git-magit-status-fullscreen t)
 (markdown :variables markdown-live-preview-engine 'vmd)
 org
 (shell :variables shell-default-height 30 shell-default-position 'bottom)
 spell-checking
 (syntax-checking :variables syntax-checking-enable-tooltips nil)
 (version-control :variables version-control-diff-tool 'diff-hl)
 osx colors idris
 (haskell :variables haskell-completion-backend 'intero haskell-enable-hindent-style "johan-tibell")
 (latex :variables latex-build-command "LatexMk")
 (evil-snipe :variables evil-snipe-enable-alternate-f-and-t-behaviors t)
 ocaml lua sml yaml
 (c-c++ :variables c-c++-enable-clang-support t)
 proof-general)
```


#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (wrong-type-argument stringp ("Spacemacs documentation" . "DOCUMENTATION.org"))
  string=(("Spacemacs documentation" . "DOCUMENTATION.org") "CONTRIBUTING.org")
  (if (string= candidate "CONTRIBUTING.org") (concat spacemacs-start-directory candidate) (concat spacemacs-docs-directory candidate))
  (let ((file (if (string= candidate "CONTRIBUTING.org") (concat spacemacs-start-directory candidate) (concat spacemacs-docs-directory candidate)))) (cond ((equal (file-name-extension file) "md") (condition-case nil (save-current-buffer (set-buffer (find-file-noselect file)) (gh-md-render-buffer) (spacemacs/kill-this-buffer)) ((debug find-file) file))) ((equal (file-name-extension file) "org") (spacemacs/view-org-file file "^" (quote all))) (t (find-file file))))
  ivy-spacemacs-help//documentation-action-open-file(("Spacemacs documentation" . "DOCUMENTATION.org"))
  ivy-call()
  ivy-read("Spacemacs Documentation: " (("Spacemacs documentation" . "DOCUMENTATION.org") ("How to contribute to Spacemacs" . "CONTRIBUTING.org") ("Vim users migration guide" . "VIMUSERS.org") ("Quick start guide for Spacemacs" . "QUICK_START.org") ("Tips on writing layers for Spacemacs" . "LAYERS.org") ("Spacemacs FAQ" . "FAQ.org") ("Spacemacs conventions" . "CONVENTIONS.org")) :action ivy-spacemacs-help//documentation-action-open-file)
  ivy-spacemacs-help-docs(nil)
  funcall-interactively(ivy-spacemacs-help-docs nil)
  call-interactively(ivy-spacemacs-help-docs nil nil)
  command-execute(ivy-spacemacs-help-docs)
```

**Comment by d12frosted** [comment]
@bixuanzju I can confirm this. Haha, total disaster with some fixes 😸 💃

**Comment by d12frosted** [comment]
PR #6517 fixes this issue but seems like it wasn't pushed, though marked as merged 😸 

@bixuanzju 💯 many thanks for finding this out. Now that fix is in develop.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
